core/common/src/main/java/alluxio/network/protocol/databuffer/DataFileChannelV2.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol.databuffer;
+
+import com.google.common.base.Preconditions;
+import io.netty.channel.DefaultFileRegion;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+/**
+ * A DataBuffer with the underlying data being a {@link FileChannel}.
+ */
+public final class DataFileChannelV2 implements DataBuffer {
```

Comment:
RPCMessageEncoder#encode calls getNettyOutput which transfers data from the file region to network.